### PR TITLE
Openai update

### LINKED ISSUE #54: OpenAI 1.0.0+ updates
## 🐛 Bug Report
OpenAI 1.0.0+ doesn't record events

### PR DESCRIPTION
**Pull request by areibman**
## 📥 Pull Request

Fixed the versioning issue where OpenAI libraries > 1.0.0 would crash. We now have handlers for both

### CODE REVIEW

**Review by areibman — COMMENTED** [review]
Does not work. Need to test against openai v1+ and retry

File: agentops/llm_tracker.py [hunk 1]
```diff
@@ -41,90 +51,257 @@ def handle_stream_chunk(chunk):
...
-        for method_path in self.SUPPORTED_APIS[api]:
-            method_parts = method_path.split(".")
-            original_method = functools.reduce(getattr, method_parts, module)
-            new_method = self._override_method(api, original_method)
+            # If openai 1.0.0+ is specified, patch the completions methods
+            if api == 'openai':
+                module_version = parse(module.__version__)
+                if module_version >= parse('1.0.0'):
+                    self.override_openai_v1_completion()
+                    self.override_openai_v1_async_completion()
```
Comment (siyangqiu) [inline agentops/llm_tracker.py]: Not all packages have a module.__version__. Make sure you are handling that case

File: tests/test_session.py [hunk 2]
```diff
@@ -166,7 +166,11 @@ def test_function_call(self, mock_req):
         prompt = 'prompt'
 
         @self.client.record_action(event_name=self.event_type)
```
Comment (siyangqiu) [inline tests/test_session.py]: Complete the merge?

File: tests/test_session.py [hunk 3]
```diff
@@ -177,5 +181,28 @@ def foo(prompt=prompt):
         assert len(mock_req.request_history) == 1
         request_json = mock_req.last_request.json()
         assert request_json['events'][0]['action_type'] == 'action'
+<<<<<<< Updated upstream
+=======
+        assert request_json['events'][0]['prompt'] == None
```
Comment (siyangqiu) [inline tests/test_session.py]: complete the merge

File: requirements.txt [hunk 4]
```diff
@@ -2,4 +2,5 @@ pytest==7.4.0
 requests==2.31.0
 requests-mock==1.11.0
 pydantic==2.4.2
-langchain==0.0.350
\ No newline at end of file
+packaging==23.1
+langchain==0.0.350
```
Comment (siyangqiu) [inline requirements.txt]: Is there a concept of a peer dependency in python

**Review by siyangqiu — APPROVED** [review]
a couple of fixes, but otherwise looks good

**Pull request merged**